controllers/pkg/reconcilers/repository/reconciler.go

```diff
@@ -0,0 +1,229 @@
+			r.l.Error(err, "cannot delete repo in git server")
+			return ctrl.Result{Requeue: true, RequeueAfter: 5 * time.Second}, errors.Wrap(r.Status().Update(ctx, cr), errUpdateStatus)
+		}
+
+		if err := r.finalizer.RemoveFinalizer(ctx, cr); err != nil {
+			r.l.Error(err, "cannot remove finalizer")
+			cr.SetConditions(infrav1alpha1.Failed(err.Error()))
+			return ctrl.Result{Requeue: true, RequeueAfter: 5 * time.Second}, errors.Wrap(r.Status().Update(ctx, cr), errUpdateStatus)
+		}
+
+		r.l.Info("Successfully deleted resource")
+		return ctrl.Result{Requeue: false}, nil
+	}
+
+	// add finalizer to avoid deleting the repo w/o it being deleted from the git server
+	if err := r.finalizer.AddFinalizer(ctx, cr); err != nil {
+		r.l.Error(err, "cannot add finalizer")
+		cr.SetConditions(infrav1alpha1.Failed(err.Error()))
+		return ctrl.Result{Requeue: true, RequeueAfter: 5 * time.Second}, errors.Wrap(r.Status().Update(ctx, cr), errUpdateStatus)
+	}
+
+	// upsert repo in git server
+	if err := r.upsertRepo(ctx, giteaClient, cr); err != nil {
+		return ctrl.Result{RequeueAfter: 5 * time.Second}, errors.Wrap(r.Status().Update(ctx, cr), errUpdateStatus)
+	}
+	cr.SetConditions(infrav1alpha1.Ready())
+	return ctrl.Result{}, errors.Wrap(r.Status().Update(ctx, cr), errUpdateStatus)
+}
+
+func (r *Reconciler) upsertRepo(ctx context.Context, giteaClient *gitea.Client, cr *infrav1alpha1.Repository) error {
+	u, _, err := giteaClient.GetMyUserInfo()
+	if err != nil {
+		r.l.Error(err, "cannot get user info")
+		cr.SetConditions(infrav1alpha1.Failed(err.Error()))
+		return err
+	}
+
+	_, _, err = giteaClient.GetRepo(u.UserName, cr.GetName())
+	if err != nil {
+		// create repo
+		createRepo := gitea.CreateRepoOption{Name: cr.GetName()}
+		if cr.Spec.Description != nil {
+			createRepo.Description = *cr.Spec.Description
+		}
+		if cr.Spec.Private != nil {
+			createRepo.Private = *cr.Spec.Private
+		}
+		if cr.Spec.IssueLabels != nil {
+			createRepo.IssueLabels = *cr.Spec.IssueLabels
+		}
+		if cr.Spec.Gitignores != nil {
+			createRepo.Gitignores = *cr.Spec.Gitignores
+		}
+		if cr.Spec.License != nil {
+			createRepo.License = *cr.Spec.License
+		}
+		if cr.Spec.Readme != nil {
+			createRepo.Readme = *cr.Spec.Readme
+		}
+		if cr.Spec.DefaultBranch != nil {
+			createRepo.DefaultBranch = *cr.Spec.DefaultBranch
+		}
+		if cr.Spec.TrustModel != nil {
+			createRepo.TrustModel = gitea.TrustModel(*cr.Spec.TrustModel)
+		}
+		createRepo.AutoInit = true
+		r.l.Info("repository", "config", createRepo)
+
+		repo, _, err := giteaClient.CreateRepo(createRepo)
+		if err != nil {
+			r.l.Error(err, "cannot create repo")
+			// Here we dont provide the full error sicne the message change every time and this will retrigger
+			// a new reconcile loop
+			cr.SetConditions(infrav1alpha1.Failed("cannot create repo"))
+			return err
+		}
+		r.l.Info("repo created", "name", cr.GetName())
+		cr.Status.URL = &repo.CloneURL
+	} else {
```

Comment:
if we `return nil` just before this else, we can get rid of the else and the indentation associated with it for the rest of the method - there is nothing else left in the method except this else clause.

Reply:
changed